text/0401-coercions.md

```diff
@@ -321,18 +321,26 @@ following holds:
 
 * `e` has type `T` and `T` coerces to `U`;
 
-* `e` has type `*T` and `U` is `*U_0` (i.e., between any raw pointers);
+* `e` has type `*T`, `U` is `*U_0`, and either `U_0: Sized` or
+    unsize_kind(`T`) = unsize_kind(`U_0`);
```

Comment:
I.e. a fat pointer (`*const Trait`) can be truncated to thin pointer (`*const u8`)? I don't think it should be done with `as`. Fat pointers definitely should have some methods for extracting their components without shady transmutes, but `as` should convert only between pointers of the same kind IMO.